apps/web/controllers/authentication.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Web
+  module Controllers
+    module Authentication
+      def self.included(action)
+        action.class_eval do
+          expose :current_user
+          before :authenticate!
+        end
+      end
+
+      def current_user
+        @current_user ||= warden&.user
+      end
+
+      private
+
+      def warden
+        params.env['warden']
+      end
+
+      def authenticate!
+        return if @skip_authentication
+
+        authenticate_current_user!
+
+        return if @skip_signing_address_authentication
```

Comment:
This will never return as the instance variable doesn't appear to be defined anywhere